src/main/groovy/com/deploygate/gradle/plugins/internal/http/ApiClient.java

```diff
@@ -1,250 +1,47 @@
-    }
-
-    @NotNull
-    @DeployGateInternal
-    @Deprecated
-    public static String getEndpoint() {
-        return sEndpoint;
-    }
-
-    @VisibleForTesting
-    static void clear() {
-        sEndpoint = Config.getDEPLOYGATE_ROOT();
-        sInstance = null;
-    }
-
-    @NotNull
-    public static ApiClient getInstance() {
-        if (sInstance != null) {
-            return sInstance;
-        }
-
-        synchronized (LOCK) {
-            if (sInstance != null) {
-                return sInstance;
-            }
-
-            sInstance = new ApiClient(sEndpoint);
-        }
-
-        return sInstance;
-    }
-
     @NotNull
     private final HttpClient httpClient;
-    @NotNull
-    private final String endpoint;
-
-    @VisibleForTesting
-    ApiClient(@NotNull String endpoint) {
-        this.endpoint = endpoint;
-
-        List<BasicHeader> headers = new ArrayList<>();
-        headers.add(new BasicHeader("X-DEPLOYGATE-CLIENT-ID", "gradle-plugin/" + Config.getVERSION_CODE()));
-        headers.add(new BasicHeader("X-DEPLOYGATE-CLIENT-VERSION-NAME", Config.getVERSION() + "-" + Config.getVERSION_NAME()));
-        headers.add(new BasicHeader("X-DEPLOYGATE-GRADLE-PLUGIN-AGP-VERSION", String.valueOf(AndroidGradlePlugin.getVersion())));
 
-        RequestConfig requestConfig = RequestConfig.custom().
-                setExpectContinueEnabled(true).
-                setMaxRedirects(3).
-                setConnectionRequestTimeout(10, TimeUnit.MINUTES).
-                setDefaultKeepAlive(10, TimeUnit.MINUTES).
-                setResponseTimeout(10, TimeUnit.MINUTES).
-                build();
+    @NotNull
+    private final Credentials credentials;
 
-        this.httpClient = HttpClientBuilder.create().
-                useSystemProperties().
-                setUserAgent("gradle-deploygate-plugin/" + Config.getVERSION()).
-                setDefaultHeaders(headers).
-                setDefaultRequestConfig(requestConfig).
-                build();
+    /**
+     * @param httpClient should be immutable
+     * @param credentials a credential of those requests
+     */
+    ApiClient(@NotNull HttpClient httpClient, @NotNull Credentials credentials) {
+        this.httpClient = httpClient;
+        this.credentials = credentials;
     }
 
     /**
      * Upload the application file to the app owner space
      *
-     * @param appOwnerName an app owner name
-     * @param apiToken an authorization token
-     * @param request a request to the server that must contain a file
+     * @param request      a request to the server that must contain a file
      * @return a successful response that contains a typed json.
      * @throws HttpResponseException is thrown if a request is an error inclduing 4xx and 5xx
-     * @throws NetworkFailure is thrown if a network trouble happens
+     * @throws NetworkFailure        is thrown if a network trouble happens
```

Comment:
unnecessary spaces?

Reply:
This is from the code style and aligned with the HttpResponseException's line. Anyway, we should have the formatter. 🤷